actor/builtin/miner/miner.go

```diff
@@ -560,11 +560,20 @@ func (ma *Actor) CommitSector(ctx exec.VMContext, sectorID uint64, commD, commR,
 
 		state.ActiveCollateral = state.ActiveCollateral.Add(collateral)
 
-		if state.Power.Equal(types.NewBytesAmount(0)) {
+		// Case 1: if the miner has never added a sector before start
+		// ther proving period now.  Every miner's first PoSt will be
+		// over a single sector.
+		//
+		// Case 2: if the miner is adding sectors during genesis
+		// construction all committed sectors accumulate in their
+		// proving set.  This is a bit of a hack but it is fairly
+		// clean and allows us to add power immediately in genesis
+		// with commitSector and submitPoSt calls without adding
+		// additional special casing for bootstrappers.
+		if state.SectorCommitments.Size() == 0 || ctx.BlockHeight().Equal(types.NewBlockHeight(0)) {
+			state.ProvingSet = state.ProvingSet.Add(sectorID)
```

Comment:
Blocking: this muddies the semantics of ProvingSet. Before this change, it identified the set of sectors a miner has submitted at least one proof for (and not yet stopped/failed), and so was a proxy for power. Now it can be non-empty when a miner has no power.

I'm ok with bootstrap miners having a special case, but not all miners. Shouldn't boostrap miners have a `state.Power = xxx` here?

Reply:
> Before this change, it identified the set of sectors a miner has submitted at least one proof for

This is not true.  Before this change the proving set identifies the set of sectors the miner is required to submit a PoSt for over the current proving period.  The fact that it is non-empty when a miner has no power is expected during the edge case when an actor first adds power.

This non-bootstrap case is coming from [the spec](https://github.com/filecoin-project/specs/blob/master/actors.md#commitsector).  If we want to improve these ideas I think filing a spec issue or spec PR is the right way forward.

> Shouldn't boostrap miners have a state.Power = xxx here?

This is probably a good idea eventually.  I wrote it this way because the change required very little special casing.  I have a feeling we should eventually be crafting genesis blocks without calling actor code at all (and setting `state.Power = xxx` in gengen code) in order to avoid lugging around genesis special casing during live execution.

Reply:
@ZenGround0's reading of the spec is consistent with my understanding. 

It's probably worth noting that checking `SectorCommitments` size instead of `ProvingSet` is inconsistent with the spec. Given the logic implemented here and the fact that SectorCommitments are a superset of ProvingSet, `SectorCommitments.Size() == 0` is equivalent statement to `ProvingSet.Size() == 0`, but it'd be nice if developers didn't have to work that out. If it's unreasonable to get the `ProvingSet` size, the equivalence should be justified in a comment.

Reply:
@acruikshank that's a good point.  It was just a mistake and will be easy to fix to match the spec.

Reply:
Before this change, `ProvingSet` was only updated in SubmitPost so could not possibly have referenced a sector not yet proven. I am reading what the code does, not what the spec suggests it should do.

Please update the comment on the state field `ProvingSet` to clearly state the "required to prove" semantics you are now implementing. Delete the sentence "It is only updated when a PoSt is submitted".

Also please update the comment for `SectorCommitments` to account for the fact that sectors are dropped from this collection when "done".

While you're there, update the comment on `Power` to also not be misleading: it's proven, not committed, storage.